src/main/java/pau/task/TaskList.java

```diff
@@ -0,0 +1,247 @@
+package pau.task;
+
+import pau.exception.DeadlineNoEndException;
+import pau.exception.NoDescException;
+import pau.exception.NoSuchTaskException;
+
+import java.util.ArrayList;
+import java.util.Scanner;
+
+/**
+ * Stores the tasks in an ArrayList;
+ */
+public class TaskList {
+
+    /**
+     * The list where the tasks are stored.
+     */
+    private ArrayList<Task> taskList;
+
+    /**
+     * Constructs a new ArrayList for the tasks.
+     */
+    public TaskList() {
+        this.taskList = new ArrayList<>();
+    }
+
+    /**
+     * Retrieves size of the task list.
+     *
+     * @return Size of the task list.
+     */
+    public int listSize() {
+        return this.taskList.size();
+    }
+
+    /**
+     * Retrieves the task at a certain index in the task list.
+     *
+     * @param index The index of the task to be retrieved.
+     * @return The task at the index.
+     */
+    public Task getTask(int index) {
+        return taskList.get(index);
+    }
+
+    /**
+     * Prints out a list of the tasks the user has.
+     */
+    public void checkList() {
+        if (this.listSize() == 0) {
+            int starEyesEmoji = 0x1F929;
+            System.out.println("yay you don't have anything to do" + new String(Character.toChars(starEyesEmoji)));
+        } else {
+            System.out.println("sian you still have to complete these:");
+            for (int i = 0; i < this.listSize(); i++) {
+                Task curr = this.taskList.get(i);
+                System.out.println((i + 1) + ". " + curr.toString());
+            }
+        }
+    }
+
+    /**
+     * Marks a task as done.
+     *
+     * @param input The task the user wants to mark as done.
+     */
+    public void markTask(String input) {
+        int starEyesEmoji = 0x1F929;
+        System.out.println("good job, you've completed a task! You're so productive!" + new String(Character.toChars(starEyesEmoji)));
+        String parts[] = input.split(" ");
+
+        int taskNo = Integer.parseInt(parts[1]);
+        Task checkedTask = this.taskList.get(taskNo - 1);
+
+        checkedTask.markAsDone();
+        System.out.println(checkedTask.toString());
+    }
+
+    /**
+     * Unmarks a task.
+     *
+     * @param input The task the user wants to unmark.
+     */
+    public void unMarkTask(String input) {
+        String parts[] = input.split(" ");
+        int taskNo = Integer.parseInt(parts[1]);
+        Task checkedTask = this.taskList.get(taskNo - 1);
+
+        checkedTask.markAsUndone();
+        System.out.println("why are you not going to " + checkedTask.description + "? remember to do it later!");
```

Comment:
Might lead to grammatically weird sentences if you print it with the description lol (small nitpick)